packages/dds/tree/docs/wip/Modules.md

```diff
@@ -0,0 +1,192 @@
+| `SharedTree`   | Set of field kinds (includes ops for these kinds), history handling policy (keep none, keep main, keep timeline, keep branches), indexes, Rebaser |                                                                                                             |                                                         | `LazyPageTree`, `Forest`, `Rebaser` |
+
+## Feature Libraries
+
+There are libraries which plug into the Core Libraries to provide specializations, either for performance, or compatibility with specific tools or use-cases.
+Lists here are examples of options we could provide, and are not intended to be exhaustive, or all things that would be worth implementing.
+
+-   Chunk Representations: Forest at least one chunk representation to actually store data. Several are provided for performance reasons:
+
+    -   IndirectNode: Single node chunk: most general, supports all functionality (generally want to always include this one)
+    -   UniformSequence: Highly compressed based on tree shape
+
+-   Schema importers: We will likely want at least one tool that takes input schema files and generates the types used by the `Schema` library.
+    Some options are listed here, but little thought has gone into this list:
+
+        -   [json schema](https://json-schema.org/)
+        -   graphQL
+        -   typescript DSL (like [TypeBox](https://www.npmjs.com/package/@sinclair/typebox)) that supports runtime and compile type typing without code gen.
+
+-   Schema libraries: We likely want to provide some useful schema, using and compatible with some importer (above):
+
+    -   json (some way to embed json into our trees)
+    -   xml
+    -   primitives (somehow we need to express primitive types / values. This can be in the form of a schema library, which may or may not be special compared to normal schema libraries)
+
+-   Schematize configurations:
+    Likely the only schematize configuration we will want is one for using it with `Schema` and `EditableTree`, but other options are possible.
+    Likely will want optional fast path for when the `EditableTree` is backed by a `Forest` using shape based compression: may be able to skip some or all validation for whole subtrees this way.
+
+-   Schematize Types: A typescript type-meta-function that subsets the `EditableTree` API to be schema aware (for both reading and editing)
+
+-   Normalizers: Libraries to convert EditableTrees into representations ideal for use in specific frameworks/platforms/tool-kits. Like schematize, may have a fast-path for specific compressed data representations.
+
+    -   react
+    -   custom?
+
+-   Rebaser ChangeSet Representations: There are a lot of different aspects to this
```

Comment:
Is "Rebaser" really meant to be here?

Reply:
Yes. This is specifically referring to "ChangeSet Representation" from the Rebaser row's "Parameterized Over" column. I've made it possessive: maybe thats better?